### DISCUSSION TITLE
How could I execute multiple conversion with ffmpeg in Nushell?

### DISCUSSION THREAD

**Discussion by otakutyrant** [nushell/nushell#11558] [Q&A] [answered]
Suppose I have only `input.mp3`, `ffmpeg -f lavfi -i color=c=blue:s=640x320 -i input.mp3 -shortest -fflags +shortest output.mp4` works.

As for all mp3 files in the current directory, I have made up a command: `ls | where name =~ "mp3" | get name | each { |it| ffmpeg -f lavfi -i color=c=blue:s=640x320 -i $it -shortest -fflags +shortest $it} `, but I do not know how to do modify the latest `$it` so that the converted files ends with mp4 suffix.

**Comment by hustcer** [comment, marked as answer]
```nu
ls | where name =~ "mp3" | get name | each { $in | path parse | get stem } | each {|it| ffmpeg -f lavfi -i color=c=blue:s=640x320 -i $'($it).mp3' -shortest -fflags +shortest $'($it).mp4' }

# OR
ls | where name =~ "mp3" | get name | each { |it| ffmpeg -f lavfi -i color=c=blue:s=640x320 -i $it -shortest -fflags +shortest ($it | str replace '.mp3' '.mp4') }
```

**Reply by otakutyrant** [reply]
You have my `[thanks appreciations gratitudes]`.
